src/backend/d3d12/CommandBufferD3D12.cpp

```diff
@@ -275,57 +276,41 @@ namespace d3d12 {
                         commands.NextCommand<BeginRenderSubpassCmd>();
                         const auto& subpassInfo = currentRenderPass->GetSubpassInfo(currentSubpass);
 
-                        uint32_t attachmentCount = currentRenderPass->GetAttachmentCount();
                         bool usingBackbuffer = false; // HACK(enga@google.com): workaround for not having depth attachments
 
-                        DescriptorHeapHandle rtvHeap = device->GetDescriptorHeapAllocator()->AllocateCPUHeap(D3D12_DESCRIPTOR_HEAP_TYPE_RTV, attachmentCount);
-                        uint32_t rtvIndex = 0;
-
+                        uint32_t rtvCount = 0;
+                        D3D12_CPU_DESCRIPTOR_HANDLE rtvHandles[kMaxColorAttachments];
                         for (uint32_t index : IterateBitSet(subpassInfo.colorAttachmentsSet)) {
                             uint32_t attachment = subpassInfo.colorAttachments[index];
                             const auto& attachmentInfo = currentRenderPass->GetAttachmentInfo(attachment);
 
-                            ComPtr<ID3D12Resource> texture;
-                            if (auto textureView = currentFramebuffer->GetTextureView(attachment)) {
-                                texture = ToBackend(textureView->GetTexture())->GetD3D12Resource();
-                            } else {
+                            D3D12_CPU_DESCRIPTOR_HANDLE rtvHandle = currentFramebuffer->GetRTVHandle(attachment);
+                            rtvHandles[rtvCount++] = rtvHandle;
+                            if (!currentFramebuffer->GetTextureView(attachment)) {
```

Comment:
This handling of the backbuffer could be hidden in Framebuffer so that the code here is almost the final version (except the clear hack).